src/java.base/windows/classes/sun/nio/fs/WindowsFileStore.java

```diff
@@ -232,12 +241,20 @@ public boolean equals(Object ob) {
         if (!(ob instanceof WindowsFileStore))
             return false;
         WindowsFileStore other = (WindowsFileStore)ob;
-        return root.equals(other.root);
+        if (root.equals(other.root))
+            return true;
+        if (volType == DRIVE_FIXED && other.volumeType() == DRIVE_FIXED)
+            return root.equalsIgnoreCase(other.root);
+        return false;
     }
 
     @Override
     public int hashCode() {
-        return root.hashCode();
+        if (hashCode == 0) { // Don't care about race
+            hashCode = volType == DRIVE_FIXED ?
```

Comment:
minor nit but would be a bit easier to read if changed to 
hashCode = (volType == DRIVE_FIXED) ? ...